tests/algorithms/test_bc.py

```diff
@@ -212,6 +212,66 @@ def test_that_bc_improves_rewards(
     )
 
 
+def test_gradient_accumulation(
+    cartpole_venv: vec_env.VecEnv,
+    rng,
+    pytestconfig,
+):
+    demonstrations = make_expert_transition_loader(
+        cache_dir=pytestconfig.cache.makedir("experts"),
+        batch_size=6,
+        expert_data_type="transitions",
+        env_name="seals/CartPole-v0",
+        rng=None,
+        num_trajectories=17,
+    )
+
+    seed = rng.integers(2**32)
+
+    th.manual_seed(seed)
+    bc_trainer1 = bc.BC(
+        observation_space=cartpole_venv.observation_space,
+        action_space=cartpole_venv.action_space,
+        batch_size=12,
+        demonstrations=demonstrations,
+        custom_logger=None,
+        rng=None,
+    )
+
+    th.manual_seed(seed)
+    bc_trainer2 = bc.BC(
+        observation_space=cartpole_venv.observation_space,
+        action_space=cartpole_venv.action_space,
+        batch_size=12,
+        minibatch_size=3,
+        demonstrations=demonstrations,
+        custom_logger=None,
+        rng=None,
+    )
+
+    for _ in range(5):
+        seed = rng.integers(2**32)
+
+        th.manual_seed(seed)
+        bc_trainer1.train(n_batches=1)
+
+        th.manual_seed(seed)
+        bc_trainer2.train(n_batches=1)
+
+        # Note: due to numerical instability, the models are
+        # bound to diverge at some point, but should be stable
+        # over the short time frame we test over; however, it is
+        # theoretically possible that with very unlucky seeding,
+        # this could fail.
+        assert all(
+            th.allclose(p2, p2, atol=3e-8)
```

Comment:
Typo -- `p2` should always be close to `p2`! (I'll fix this; requires increasing `atol` though.)

Reply:
Yeah, not sure how I missed that. `atol=1e-5` seems a bit high though, did you try increasing `rtol` instead? Default values are `atol=1e-8, rtol=1e-5` (from https://pytorch.org/docs/stable/generated/torch.allclose.html) I would hope that increasing them to `atol=1e-7, rtol=1e-4` would be enough.

Reply:
I didn't try increasing `rtol`, thanks for the suggestion! Unfortunately fails sometimes even with `rtol=1e-3`, so I think `atol=1e-5` is our best bet. The need for something this large does make me a bit suspicious of the code, although float32 precision is only around `5e-8` so if the tensor has 20 elements then getting an error of up to `1e-6` is possible even in the best-case scenario.

Reply:
Just as a sanity check, did you increase them both at the same time?

If it still behaves the same, then just do what you think is best. But let me know if you think it warrants a bigger change in case I can help.